client/verta/verta/registry/_check_model_dependencies.py

```diff
@@ -60,14 +61,26 @@ def _check_model_dependencies(
         check_model_dependencies(model_cls=MyModelClass, environment=env)
 
     """
+    env_packages: Set[str] = {parse_req_spec(e)[0] for e in environment.requirements}
     detected_modules: Set[str] = md.class_module_names(model_cls)
-    detected_packages: Set[str] = md.package_names(detected_modules)
-    env_packages: Set[str] = { parse_req_spec(e)[0] for e in environment.requirements }
+    detected_packages: Dict[str, List[str]] = md.package_names(detected_modules)
+
+    missing_packages: Dict[str, List[str]] = {
+        mod: pkgs
+        for mod, pkgs in detected_packages.items()
+        if not env_packages.intersection(set(pkgs))
+           and mod not in env_packages  # namespace package
```

Comment:
It's a bit surprising to me that we'd need to check for `mod not in env_packages` if we're already checking `pkg`s 🤔 Are you sure we need this?

Reply:
Yeah, once again `google` is the example problem child.  A user could just do `Python(['google'])`.  The result would be the installation of ~5 different packages, none of which is named "google".  So when we compare packages, we have to add the plain "google" option to the list of acceptable package options for supporting the given module.  We could move that logic to `package_names()` instead of here...

Reply:
Hmmm I'm still not sure about this. If `md.package_names()` says "The root module `'google'` is supplied by the pip-installable packages `['googleapis-common-protos', 'google-cloud-bigquery', 'protobuf', 'google-cloud-core', 'google-api-core', 'google-resumable-media', 'google-auth']`", and `env_packages` contains pip-installable package names, I don't quite see the purpose of also checking for the root module name.

> The result of `pip install google` would be the installation of ~5 different packages

This isn't true.

Reply:
Just installed `google` on a clean virtual env and you are absolutely right.  `pip install google` just installs beautifulsou4 and a google search module.  I don't know how my main modeldb virtual env has a package named "google" in `packages_distributions()`, but it does and it's causing the unit tests to fail when I don't catch it.

Reply:
In any case, passing "google" as a dependency to the env would not work to support the modules we're using in the test case.  I'll pull this last line of the comprehension out and adjust the test env to use any of the appropriate google packages.

Reply:
Done.